engine/table/src/main/java/io/deephaven/engine/table/impl/sources/LongArraySource.java

```diff
@@ -598,29 +598,18 @@ public <R> void fillPrevChunk(
                 // region conditionalCopy
                 long[] baseInput = (long[]) getBlock(blockNo);
                 long[] overInput = (long[]) getPrevBlock(blockNo);
-                effectiveContext.copyKernel.conditionalCopy(destination, baseInput, overInput,
-                        inUse, srcOffset, destOffset.intValue(), length);
+                final boolean flipBase = false;
 
-                int bitsSet = 0;
-                final int bitsetLen = (length + 63) >> 6;
-                final int bitsetOffset = srcOffset >> 6;
-                for (int i = 0; i < bitsetLen; ++i) {
-                    bitsSet += Long.bitCount(inUse[i + bitsetOffset]);
-                }
-                final int totalBits = bitsetLen << 6;
-                final boolean flipBase = bitsSet > totalBits / 2;
+                final int srcEndOffset = srcOffset + length;
+                int nextBit = CopyKernel.Utils.nextSetBit(inUse, srcOffset, srcEndOffset, flipBase);
 
-                // mem-copy from baseline
                 for (int ii = 0; ii < length; ++ii) {
-                    chunk.set(destOffset.intValue() + ii, converter.apply((flipBase ? overInput : baseInput)[srcOffset + ii]));
-                }
-
-                final int srcEndOffset = srcOffset + length;
-                for (int ii = CopyKernel.Utils.nextSetBit(inUse, srcOffset, srcEndOffset, flipBase);
-                     ii < srcEndOffset;
-                     ii = CopyKernel.Utils.nextSetBit(inUse, ii + 1, srcEndOffset, flipBase)) {
-                    chunk.set(destOffset.intValue() + ii - srcOffset,
-                            converter.apply(flipBase ? baseInput[ii] : overInput[ii]));
+                    if (ii != nextBit - srcOffset) {
+                        chunk.set(destOffset.intValue() + ii, converter.apply(baseInput[srcOffset + ii]));
```

Comment:
We have no coverage for this line, or the case where none in the block are in use.